### PR TITLE
[CN-554] Change default repo to EE when licenseKeySecret present

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

**Review by SeriyBg — APPROVED** [review]
Have two small comments, but other than that LGTM

File: controllers/hazelcast/mutate/mutate.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package mutate
+
+import (
+	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	n "github.com/hazelcast/hazelcast-platform-operator/internal/naming"
+)
+
+func MutateHazelcastSpec(h *hazelcastv1alpha1.Hazelcast) (mutated bool) {
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply controllers/hazelcast/mutate/mutate.go:8]: Ah, got it. I will change it :+1:

File: controllers/hazelcast/mutate/mutate.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package mutate
```
[the remark this answers is not in the document]
Comment (SeriyBg) [reply controllers/hazelcast/mutate/mutate.go:1]: By defaults, I mean setting the default values for some fields, for example, the values from [here](https://github.com/hazelcast/hazelcast-platform-operator/blob/eb2b5c25279222bc1d0848d3cd07e3ceeda48c20/internal/naming/constants.go#L66)

[the remark this answers is not in the document]
Comment (mtyazici) [reply controllers/hazelcast/mutate/mutate.go:1]: But we are not setting them in the code, we are using kubebuilder tags to add the defaults to OpenAPISchema to the api server handles the defaults.

Those values are only there so we can validate the default values in the tests.

[the remark this answers is not in the document]
Comment (SeriyBg) [reply controllers/hazelcast/mutate/mutate.go:1]: Oh, you are right. It has been changed 👍

**Review by cagric0 — APPROVED** [review]
LGTM!

**Pull request merged**